sensorhub: decrease magnetometer base sampling freq. from 80Hz to 64Hz

niewim19 opened the pull request:
JIRA: PP-51

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes shortly -->
Longer wait is needed for magnetometer thread to not truncate data when sampling the sensor without uisng DRDY (data ready) register check. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
We skip DRDY check as it is not important for us to achieve absolute maximum magnetometer data sampling performance. It comes out, that precise 80Hz sampling causes data loss on some axis. Thus the closest and possibly round value of 64 Hz was chosen as sampling frequency. Tested on zturn drone this fixes the problem. 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: lsm9dsxx with zturn.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

nalajcie commented:
Please move the description as to WHY the change was needed to commit message body (instead of writing it only in PR) - the commit messages stay with us forever, the PR descriptions are as good as lost after the merge :).

Side note: it could be written as `usleep((1000*1000) / 64)` so I would see 64Hz without additional comment. Also, sleep-based sampling time will always be irregular (and below expected 64Hz) but I suppose it doesn't matter in this case.

niewim19 commented:
> Please move the description as to WHY the change was needed to commit message body (instead of writing it only in PR) - the commit messages stay with us forever, the PR descriptions are as good as lost after the merge :).
> 
> Side note: it could be written as `usleep((1000*1000) / 64)` so I would see 64Hz without additional comment. Also, sleep-based sampling time will always be irregular (and below expected 64Hz) but I suppose it doesn't matter in this case.



> Please move the description as to WHY the change was needed to commit message body (instead of writing it only in PR) - the commit messages stay with us forever, the PR descriptions are as good as lost after the merge :).
> 
> Side note: it could be written as `usleep((1000*1000) / 64)` so I would see 64Hz without additional comment. Also, sleep-based sampling time will always be irregular (and below expected 64Hz) but I suppose it doesn't matter in this case.

- Moved description to preserve change cause. Thanks for that!
- changed notation to proposed and removed comment.
- Yes, I am keeping in mind that this frequency is somewhat theoretical max. Thus 80Hz was chosen at start so with processing included I was counting on good results. Comes out that more time is needed. Magnetometer data at 64Hz is still very good frequency (i think we would be good even at around 10Hz)
